src/backend/distributed/commands/statistics.c

```diff
@@ -445,19 +456,32 @@ GetExplicitStatisticsCommandList(Oid relationId)
 
 		if (alterStatisticsOwnerCommand != NULL)
 		{
-			alterStatisticsCommandList =
-				lappend(alterStatisticsCommandList,
+			explicitStatisticsCommandList =
+				lappend(explicitStatisticsCommandList,
 						makeTableDDLCommandString(alterStatisticsOwnerCommand));
 		}
 	}
 
+	/* find indexes on current relation, in case of modified stats target */
+	Relation relation = relation_open(relationId, AccessShareLock);
+	List *indexOidList = RelationGetIndexList(relation);
+
+	Oid indexId = InvalidOid;
+	foreach_oid(indexId, indexOidList)
+	{
+		/* we need alter index commands for altered targets on expression indexes */
+		List *alterIndexStatisticsCommands = GetAlterIndexStatisticsCommands(indexId);
+
+		explicitStatisticsCommandList =
+			list_concat(explicitStatisticsCommandList, alterIndexStatisticsCommands);
+	}
+
+	relation_close(relation, AccessShareLock);
+
```

Comment:
We put this part in this function but this is not really an `ALTER STATISTICS` command. It is `ALTER INDEX`.

Reply:
yes, this function's goal is to collect commands for `CREATE STATISTICS` `ALTER STATISTICS` and `ALTER INDEX SET STATISTICS` necessary for distributing a table